aten/src/ATen/core/operator_name.cpp

```diff
@@ -0,0 +1,17 @@
+#include <ATen/core/operator_name.h>
+
+namespace c10 {
+
+std::string toString(const OperatorName& opName) {
+  std::string result = opName.name;
+  if (opName.overload_name.size() != 0) {
+    result += "." + opName.overload_name;
+  }
+  return result;
+}
+
+std::ostream& operator<<(std::ostream& os, const OperatorName& opName) {
+  return os << toString(opName);
```

Comment:
I'd rather implement it the other way round - have `operator<<` do the actual work and `toString` call `operator<<` with a `std::ostringstream`. In the current implementation, `operator<<` needs to materialize the temporary string. If you do it the other way round, `operator<<` does not need to materialize it anymore an `toString` isn't slower either.